vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/data/AttributeIndex.kt

```diff
@@ -52,12 +52,14 @@ object AttributeIndex {
             .firstOrNull { it.isoDocType!!.startsWith(docType) || docType.startsWith(it.isoDocType!!) }
     }
 
-    // TODO Probably shouldn't be that permissive
-    fun resolveCredentialScheme(attributeType: String): ConstantIndex.CredentialScheme? {
-        return resolveAttributeType(attributeType)
-            ?: resolveSdJwtAttributeType(attributeType)
-            ?: resolveIsoNamespace(attributeType)
-            ?: resolveSchemaUri(attributeType)
+    /**
+     * Compares the input to all CredentialScheme identifiers and on match returns it plus its associated `ConstantIndex.CredentialRepresentation` if applicable
+     */
+    fun resolveCredential(input: String): Pair<ConstantIndex.CredentialScheme, ConstantIndex.CredentialRepresentation?>? {
```

Comment:
Where would we need such a method? Doesn't the caller know if its an SD-JWT or ISO-Credential?

Reply:
The caller would know, however a receiver of an identifier would not. To identify it using VClib one needs to repeat exactly these steps. We currently have no way of matching an identifier to a representation.